ee/tabby-webserver/src/service/auth.rs

```diff
@@ -359,16 +359,17 @@ impl AuthenticationService for DbConn {
 
         let email = client.fetch_user_email(code, credential).await?;
 
-        let user =
-            if let Some(user) = self.get_user_by_email(&email).await? {
-                user
-            } else {
-                let Some(invitation) = self.get_invitation_by_email(&email).await? else {
-                    return Err(GithubAuthError::UserNotInvited);
-                };
-                let id = self.create_user_with_invitation(email, "".to_owned(), false, invitation.id).await?;
-                self.get_user(id).await?.unwrap()
+        let user = if let Some(user) = self.get_user_by_email(&email).await? {
+            user
+        } else {
+            let Some(invitation) = self.get_invitation_by_email(&email).await? else {
+                return Err(GithubAuthError::UserNotInvited);
             };
+            let id = self
+                .create_user_with_invitation(email, "".to_owned(), false, invitation.id)
```

Comment:
that's a good question.

as I checked, the input struct for both normal `register` & `login` contain password validation with min/max length, so the api call would fail. (I bet FE should have similar checks?)

also, parsing password hash from empty string will cause `Err(PhcStringField)`, which will cause `password_verify` fail, and return error response accordingly.

so by right, user from github login can not call `register` or `tokenAuth` graphql api

Reply:
sounds good，please add some comments here to